### PR TITLE
Refactoring + some tests

### PR DESCRIPTION
**Pull request by diegosouza**
**Key changes:**

1) Moved `everything_in_lib.ex` to `lib/youtex/respective/path.ex`
2) `Youtex.TranscriptListFetcher` to `Youtex.Transcript.Fetch` and some other related
3) `Video` e `Sentence` structs introduced (Video to be more explored in the future)
4) List of transcripts **(before)**:  

```elixir
%Youtex.TranscriptList{
  items: [
    %Youtex.Transcript{
      generated: false,
      language_code: "en",
      name: "Inglês",
      url: "https://www.youtube.com/api/timedtext...",
      video_id: "lxYFOM3UJzo"
    },
    %Youtex.Transcript{...},
    ...
  ]
}
```
Simplified to:

```elixir
[
  %Youtex.Transcript{
    generated: false,
    language_code: "en",
    name: "Inglês",
    sentences: [],
    url: "https://www.youtube.com/api/timedtext..."
  },
  %Youtex.Transcript{...},
  ...
]
```
5) The transcription **(before)**:

```elixir
[
  %Youtex{
    duration: 9.3,
    start: 9.53,
    text: "I remember like my first computer was a\nPentium 100 megahertz. I would be in"
  },
  %Youtex{...},
  ...
]
```

The trascription, now with sentences:

```elixir
%Youtex.Transcript{
  generated: false,
  language_code: "en",
  name: "Inglês",
  sentences: [
    %Youtex.Transcript.Sentence{
      duration: 9.3,
      start: 9.53,
      text: "I remember like my first computer was a\nPentium 100 megahertz. I would be in"
    },
    %Youtex.Transcript.Sentence{...},
     ...
  ],
  url: "https://www.youtube.com/api/timedtext..."
}
```
**Code coverage:**

```
Percentage | Module
-----------|--------------------------
    86.67% | Youtex.Transcript.Fetch
    87.50% | Youtex.Transcript
   100.00% | Youtex
   100.00% | Youtex.HttpClient
   100.00% | Youtex.Transcript.Parser
   100.00% | Youtex.Transcript.Sentence
   100.00% | Youtex.Types
   100.00% | Youtex.Video
-----------|--------------------------
    93.75% | Total
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,12 +15,54 @@ def deps do
...
+       text: "I remember like my first computer was a\nPentium 100 megahertz. I would be in"
+     },
+     %Youtex.Transcript.Sentence{...},
+     ...
+   ],
+   url: "https://www.youtube.com/api/timedtext..."
+ }}
+```
+
+If you don't want or don't need to pattern match {:ok, data} or {:error, reason}, there are also [trailing bang](https://hexdocs.pm/elixir/1.11.4/naming-conventions.html#trailing-bang-foo) versions for every one.
```
Comment (patrykwozinski) [inline README.md:64]: 👍🏻  in point!

File: lib/youtex/transcript.ex [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
+      generated: generated(caption),
     }
   end
 
+  @spec for_language(transcript_list, language) :: transcript_found | error
+  def for_language(transcript_list, language) do
+    transcript_list
+    |> Enum.filter(&(&1.language_code == language))
+    |> List.first()
+    |> transcript_found_or_error()
```
Comment (patrykwozinski) [inline lib/youtex/transcript.ex:30]: Like this one!

File: lib/youtex/types.ex [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
 
   defmacro __using__(_opts) do
     quote do
+      @type video :: Video.t
       @type video_id :: String.t
       @type language :: String.t
 
-      @type success :: {:ok, [Youtex.t]}
+      @type transcript_list :: list(Transcript.t)
+      @type sentence_list :: list(Sentence.t)
```
Comment (patrykwozinski) [inline lib/youtex/types.ex:15]: 👍🏻

Comment (patrykwozinski) [reply lib/youtex/types.ex:15]: What's the difference between `list()` and `[]`? We can pick `list(x)` as our coding-style for this project and use only it. :)

File: lib/youtex/transcript/setence.ex [hunk 4]
```diff
@@ -0,0 +1,15 @@
+defmodule Youtex.Transcript.Sentence do
```
Comment (patrykwozinski) [inline lib/youtex/transcript/setence.ex:1]: Yep, that's the time for a dedicated struct for this. :) Good job @diegosouza

File: lib/youtex/transcript/parser.ex [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  defp raw_sentences(%{"transcript" => %{"text" => sentences}}), do: sentences
+
+  defp to_sentences(sentences), do: Enum.map(sentences, &to_sentence/1)
+
+  defp to_sentence(%{"#content" => text, "-start" => start, "-dur" => duration}) do
+    {start, _} = Float.parse(start)
+    {duration, _} = Float.parse(duration)
+    Sentence.new(text, start, duration)
+  end
```
Comment (patrykwozinski) [inline lib/youtex/transcript/parser.ex:27]: Good to see it there, not in `Sentence` so the `Sentence` area doesn't know about the incoming format. :)

File: test/youtex_test.exs [hunk 6]
```diff
@@ -0,0 +1,89 @@
+defmodule YoutexTest do
+  use ExUnit.Case, async: true
+
+  use Youtex.Types
+  alias Youtex.Transcript
+  import Youtex
+
+  @en_only_video_id "guc16D_0Imk"  # ElixirConf 2020 - José Valim
+  @en_pt_video_id "lxYFOM3UJzo" # Elixir: The Documentary
```
Comment (patrykwozinski) [inline test/youtex_test.exs:9]: ❤️

**Review by patrykwozinski — APPROVED** [review]
Do these tests run in isolation?
If not - we could create some mocks for HTTP and provide stub data. I think that might be a good next step to do.

**Comment by patrykwozinski** [comment]
Great job... really I appreciate your work!

**Comment by diegosouza** [comment]
> Do these tests run in isolation?
> If not - we could create some mocks for HTTP and provide stub data. I think that might be a good next step to do.

Unfortunately not yet. They depend on the real Youtube's request/response. I thought about it.
They're more like integration tests right now. But once we have these tests, we can reorganize things safer.

I agree with you :+1:
Next step would be to have some stubs/dummies or even minimal mocks to test parts in a more isolated way.

File: lib/youtex/types.ex (hunk 3 above)
Comment (diegosouza) [reply lib/youtex/types.ex:15]: > What's the difference between `list()` and `[]`? We can pick `list(x)` as our coding-style for this project and use only it. :)

I guess there are some small differences regarding proper/improper list or regarding emptyness, I guess. That's something I'm not sure.

Good, we'll try to keep the same approach.

File: test/youtex_test.exs (hunk 6 above)
Comment (diegosouza) [reply test/youtex_test.exs:9]: Valim is a great inspiration for all the community :clap:

**Pull request merged**